bin/rad-diff

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/env radicle
+
+;; TODO:
+;;   - docstrings
+;;   - move arg-parsing stuff into a library (and factor it out of rad-issues too)
+;;   - pretty print prs
+;;   - don't use make-chain-ref when we don't need loading
+;;   - deal with multiple patches properly (e.g. in pretty-pr)
+;;   - make and use cmd-2 pattern
+;;   - handle parsing numbers in cmd-parsing/validation logic
+;;   - be graceful about checkouts failing
+;;   - should we be using "git diff", rather than squashing and then git format-patch?
+
+
+(load! (find-module-file! "prelude.rad"))
+(load! (find-module-file! "monadic/diff.rad"))
+(load! (find-module-file! "prelude/io-utils.rad"))
+(load! (find-module-file! "prelude/cmd-parsing.rad"))
+(import prelude/validation :as 'validation)
+
+(def base-name "http://localhost:8000/chains/")
+
+(def make-name
+  (fn [chain]
+    (if (eq? (take 7 chain) "http://")
+        chain
+        (string-append base-name chain))))
+
+(def make-chain-ref
+  (fn [chain]
+    (ref (chain/load-chain! (make-name chain)))))
+
+(def help
+  (string-append
+  "rad-diff - Radicle Diff CLI
+
+   Usage:
+        rad-diff list <chain-name> [--new] [--accepted] [--rejected] [--proposed] [--retracted] [--fancy]
+        rad-diff create-chain <chain-name>
+        rad-diff propose <commit> <chain-name>
+        rad-diff [retract|accept|reject|show] <diff-no> <chain-name>
+        rad-diff comment <diff-no> <comment> <chain-name>
+        rad-diff done
+        rad-diff help
+
+     list         - Select from diffs to view or comment on
+     done         - Mark all diffs as 'seen'
+     propose      - Create a new diff in $EDITOR
+     show         - Show a diff
+     comment      - Shortcut for adding a comment
+     retract      - Retract your own diff
+     accept       - Accept a diff
+     reject       - Reject a diff
+     create-chain - Create a new diffs chain
+     help         - Print this help and exit
+
+   If <chain-name> begins with 'http://', that URL will be used. Otherwise,
+  " base-name " will be prepended to <chain-name>." ))
+
+(def cmd-parse-failure
+  (fn [error]
+    (parse-failure error help)))
+
+(def mark-last-seen
+  (fn []
+    (write-file-key! ".radicle-diffs" :last-mark-done (now!))))
```

Comment:
I think it's better to keep this in git config. Otherwise we'd have to add it to git ignore, and moreover it wouldn't work if we cd into a dir inside the repo.

Reply:
Is writing it to `git config` done in the project part? 
Here I would read out the location of `.radicle-diff` via git config then and also when creating the file?